packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/queued_item_store/drift/drift_queued_item_store.dart

```diff
@@ -30,26 +34,37 @@ class DriftDatabaseJsonStrings extends _$DriftDatabaseJsonStrings {
   Future<List<DriftJsonString>> get allJsonStrings =>
       select(driftJsonStrings).get();
 
-  /// Get [maxToGet] number of [DriftJsonString] from the database
-  /// Returns oldest first
-  Future<List<DriftJsonString>> getJsonStrings(int maxToGet) {
+  @override
+  Future<void> add(String jsonString) async {
+    await into(driftJsonStrings)
+        .insert(DriftJsonStringsCompanion(jsonString: Value(jsonString)));
+  }
+
+  @override
+  Future<Iterable<QueuedItem>> get(int maxToGet) async {
     final statement = (select(driftJsonStrings)
       ..orderBy([(v) => OrderingTerm.asc(v.id)])
       ..limit(maxToGet));
 
-    return statement.get();
+    final retrievedJsonStrings = await statement.get();
+    return retrievedJsonStrings.map(
+      (jsonString) => QueuedItem(
+        id: jsonString.id,
+        value: jsonString.jsonString,
```

Comment:
Anytime you have code like `foo.foo` that is a good indication that the naming could be clearer.

I would do the following:
- Rename DriftJsonStrings. I believe the serialization to and from json happens elsewhere in the code. The data does happen to JsonStrings, but this class doesn't know what. I would name it DriftString, or DriftQueuedItem.
-  Rename `jsonString` on the DriftJsonStrings class. Again, it happens to be a jsonString, but it could be any string. I would name it value or data.